### PR TITLE
feat(eva-icons): add eva-icons module

### LINKED ISSUE #522: Add icons module

### LINKED ISSUE #548: Bottom navigation with icons: invalid props tintColor
### Issue type

**I'm submitting a ...**  (check one with "x")

* [x] bug report?
* [ ] feature request

### Issue description

**Current behavior:**
When I add any icons to bottom tab navigation (e.g. @expo/vector-icons) it shows me 
Warning: Failed prop type: Invalid props.style key `tintColor` supplied to `Text`.

**Expected behavior:**
no warnings.

**Steps to reproduce:**
I've basically copy-pasted "With React Navigation API and usage example" from Bottom Navigation documentation and added few icons from @expo/vector-icons to the tabs. 

**Related code:**

```
import { MaterialIcons } from '@expo/vector-icons';

const WalletIcon = ({ tintColor }) => 
  <MaterialIcons color={tintColor} name="compare-arrows" size={24} />

<BottomNavigationTab title='Accounts' icon={WalletIcon} />
```

### Other information:

**OS, device, package version**
```
WIN10, Android Emulator API 29
```

### PR DESCRIPTION
**Pull request by artyorsh**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

Adds icons module and Icon component

Closes #522 
Closes #548

### CODE REVIEW

File: src/framework/ui/icon/icon.component.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+ *        onPress={this.onPress}
+ *      />
+ *    );
+ *  }
+ * }
+ * ```
+ */
+
+export class Icon<T> extends React.Component<IconProps<T>> {
+
```
Comment (32penkin) [inline src/framework/ui/icon/icon.component.tsx]: Extra line

File: package.json [hunk 2]
```diff
@@ -45,7 +45,9 @@
     "@eva-design/dss": "^1.0.1",
     "@eva-design/processor": "^1.0.1",
     "hoist-non-react-statics": "^3.2.1",
-    "lodash.merge": "^4.6.1"
+    "lodash.merge": "^4.6.1",
+    "react-native-eva-icons": "^1.1.0",
```
Comment (32penkin) [inline package.json]: Do we really need to provide these dependencies if the user didn’t install the icon module?

File: src/eva-icons/evaIcon.component.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { SvgProps } from 'react-native-svg';
+import { IconProvider } from 'react-native-ui-kitten';
+
+type IconElement = React.ReactElement<SvgProps>;
+type IconComponent = React.ComponentType<SvgProps>;
+
+export class EvaIcon implements IconProvider<SvgProps> {
+
+  constructor(private content: IconComponent) {
```
Comment (32penkin) [inline src/eva-icons/evaIcon.component.tsx]: I did not know, it is possible to initialize the class field in this way. Or I'm wrong?

File: src/framework/ui/icon/iconRegistry.component.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+ *         theme={lightTheme}>
+ *         <Application/>
+ *         <IconRegistry icons={EvaIconsPack}/>
+ *       </ApplicationProvider>
+ *     );
+ *   }
+ * }
+ * ```
+ */
+export class IconRegistry extends React.Component<IconRegistryProps> {
```
Comment (32penkin) [inline src/framework/ui/icon/iconRegistry.component.tsx]: Should it be the separate component that stores icons-pack? Also, can we change the icons-pack in runtime in this case?

**Comment by 32penkin** [comment]
Merge the master in this branch

File: package.json (hunk 2 above)
Comment (artyorsh) [reply package.json]: This is a project dependencies, not a ui module

File: src/eva-icons/evaIcon.component.tsx (hunk 3 above)
Comment (artyorsh) [reply src/eva-icons/evaIcon.component.tsx]: Yes, this is a TypeScript syntax sugar

File: src/framework/ui/icon/iconRegistry.component.tsx (hunk 4 above)
Comment (artyorsh) [reply src/framework/ui/icon/iconRegistry.component.tsx]: Yes
I've gived a detailed explanation on it's usage in a [setup guide](https://github.com/akveo/react-native-ui-kitten/blob/7de76cda8cfbbf7cabdfa8ecd0c7dafdc7e8ad92/docs/src/articles/guides/setup-icons-module.md#configure-icon-registry)

Also, take a look at [supporting multiple icon packs guide](https://github.com/akveo/react-native-ui-kitten/blob/7de76cda8cfbbf7cabdfa8ecd0c7dafdc7e8ad92/docs/src/articles/guides/setup-vector-icons.md)

[Runtime icon pack change](https://github.com/akveo/react-native-ui-kitten/blob/7de76cda8cfbbf7cabdfa8ecd0c7dafdc7e8ad92/docs/src/articles/guides/setup-vector-icons.md#choose-a-pack)

**Pull request merged**